### DISCUSSION TITLE
Additional Features

### DISCUSSION THREAD

**Discussion by darthShadow** [lostb1t/replex#24] [Ideas]
Hi,

I have been following this project with interest and feel there's a lot of potential for it.

Some additional features that you can consider adding are (all taken from [RoyXiang/plexproxy](https://github.com/RoyXiang/plexproxy)):

* Traffic control by devices
* Cross-device response caching by client type
* Disable transcoding by forcing direct play/stream
* Redirect web app to [official one](https://app.plex.tv/desktop)
* [Plaxt](https://github.com/XanderStrike/goplaxt) integration

---

Some more ideas (taken from [danopia/plex-backhaul](https://github.com/danopia/plex-backhaul) this time 😄):

* Prefetch the next video segment of the file that's being streamed
* Multiplex Video Segments over multiple threads so that it's downloaded faster

Reference for both of the above: https://github.com/danopia/plex-backhaul/issues/1

**Comment by lostb1t** [comment]
I’m mostly focused on data mangling.
Caching is high on the list. It’s a bit slow right now.

Transcoding might be a thing in the future. But now that plex defaults to maximum in the clients the need for it is less

**Comment by darthShadow** [comment]
> But now that plex defaults to maximum in the clients

It's limited to Android TVs only afaik. There was a message that it would rolled out eventually to other devices too but that was months ago and nothing so far: https://forums.plex.tv/t/default-all-clients-to-max-internet-streaming/440641/1592

**Comment by darthShadow** [comment]
The prefetch and parallel download threads may also help users who have appboxes that are hosted far away but can run a smaller binary/proxy locally. This is another long pending FR btw: https://forums.plex.tv/t/better-internet-streaming-using-multiple-parallel-connections-or-threads/236442

**Comment by darthShadow** [comment]
> data mangling

If you can manipulate the `includeRelated` parameter (by sending it to plex as set to `0` and inserting empty related data from the proxy), then its probably going to help a lot of folks. It causes a lot of issues when there is bad metadata on the Plex side: https://forums.plex.tv/t/plex-movie-issue-cpu-death-spiral/828916

**Reply by lostb1t** [reply]
I'm not totally getting wiser out of that topic. 

includeRelated on what endpoint? An library item endpoint? Do you have an example?

**Reply by darthShadow** [reply]
Yep, `includeRelated` param on the item endpoint. It weirdly gets called when starting playback of an item too and any delay or hiccup there completely prevents the playback.

Some more details here: https://forums.plex.tv/t/view-xml-and-playback-of-certain-items-makes-plex-unresponsive-includerelated-1/830514

**Reply by lostb1t** [reply]
Yeah I'm open tot that. If you can create a detailed issue I'll pick it up somwhere next week

**Reply by darthShadow** [reply]
Will do, thanks.

**Reply by darthShadow** [reply]
Done, thank you: https://github.com/sarendsen/replex/issues/26

**Comment by darthShadow** [comment]
Another long-term goal may be to have a JS/TS version of this so it can be deployed on something like Cloudflare Workers.

**Reply by darthShadow** [reply]
On further digging, this may actually not require a rewrite at all but probably some kind of integration with https://github.com/cloudflare/workers-rs
